packages/manager/apps/pci-rancher/src/pages/dashboard/updateSoftware/UpdateSoftware.component.tsx

```diff
@@ -0,0 +1,193 @@
+};
+
+const UpdateSoftware: FC<UpdateSoftwareProps> = ({
+  rancher,
+  versions,
+  onClickUpdate,
+  isUpdatePending,
+}) => {
+  const { projectId } = useParams();
+
+  const { t } = useTranslate('pci-rancher/updateSoftware');
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const currentVersion = getVersion(rancher);
+
+  const availableVersions = getLatestVersionsAvailable(rancher, versions);
+  const currentVersionDetails = versions?.find(
+    (version) => version.name === currentVersion,
+  );
+  const hrefRancherById = useHref(getRancherByIdUrl(projectId, rancher?.id));
+
+  const [selectedVersion, setSelectedVersion] = useState(currentVersion);
+
+  useEffect(() => {
+    if (versions?.length > 0) {
+      setSelectedVersion(versions[versions.length - 1].name);
+    }
+  }, [versions]);
+
+  return (
+    <DashboardLayout
+      breadcrumb={
+        <Breadcrumb
+          items={[
+            {
+              label: t('updateSoftwareRancherBreadcrumb'),
+            },
+          ]}
+        />
+      }
+      content={
+        <div className="max-w-4xl">
+          <div className="my-4">
+            <LinkIcon
+              href={hrefRancherById}
+              text={t('updateSoftwareRancherPreviousButton')}
+              iconName={ODS_ICON_NAME.ARROW_LEFT}
+              slot="start"
+            />
+          </div>
+          <Subtitle>{t('updateSoftwareRancherTitle')}</Subtitle>
+          {currentVersionDetails && versions?.length > 0 && (
+            <VersionTable
+              versions={availableVersions}
+              selectedVersion={selectedVersion}
+              setSelectedVersion={setSelectedVersion}
+              currentVersion={currentVersionDetails}
+            />
+          )}
+          <div>
+            <OsdsText color={ODS_THEME_COLOR_INTENT.text}>
+              {t('updateSoftwareRancherDurationInfo')}
+            </OsdsText>
+          </div>
+          <div className="mt-5">
+            <OsdsText color={ODS_THEME_COLOR_INTENT.text}>
+              {t('updateSoftwareRancherImpact')}
+            </OsdsText>
+          </div>
+          <div className="mt-5">
+            <OsdsText color={ODS_THEME_COLOR_INTENT.text}>
+              {t('updateSoftwareRancherUpgradeInfo')}
+            </OsdsText>
+          </div>
+          <div className="mt-6 max-w-xs">
+            <OsdsButton
+              size={ODS_BUTTON_SIZE.sm}
+              color={ODS_THEME_COLOR_INTENT.primary}
+              onClick={() => setShowConfirmModal(true)}
+            >
+              {t('updateSoftwareRancherCta')}
+            </OsdsButton>
+          </div>
+          {showConfirmModal && (
+            <UpdateSoftwareModal
+              version={versions[0]}
+              isUpdatePending={isUpdatePending}
+              onConfirmUpdated={() => onClickUpdate(selectedVersion)}
+              onClose={() => setShowConfirmModal(false)}
+            />
+          )}
+        </div>
```

Comment:
**nitpick:** Can this be a constant? It looks hard to understand that it's a prop.